openapi.yaml

```diff
@@ -132,147 +457,273 @@ securityDefinitions:
+        example: 5
+      gramsSalt:
+        type: integer
+        description: the amount of salt to use, in grams
+        example: 15
+      gramsSugar:
+        type: integer
+        description: the amount of sugar to use, in grams
+        example: 20
+      gramsOliveOil:
+        type: integer
+        description: the amount of olive oil to use, in grams
+        example: 50
+      bulkFermentTime:
+        type: integer
+        description: the time in minutes to spend on the bulk ferment
+        example: 60
+      proofTime:
+        type: integer
+        description: the time in minutes to spend on the proof
+        example: 15
+      specialInstructions:
+        type: string
+        description: any additional instructions not captured in the main object
+        example: keep the dough in the fridge during the bulk ferment
+  ErrorResult:
+    properties:
+      errors:
+        type: array
+        items:
+          $ref: "#/definitions/ErrorObject"
   ErrorObject:
     properties:
       status:
         type: string
         description: HTTP status code
-        example: '123'
+        example: "123"
       title:
         type: string
         description: A short, user readable summary of the error
         example: Not Found
       code:
         type: string
         description: An application-specific error code
-        example: '1234'
+        example: "1234"
       detail:
         type: string
         description: A long description of the error that may contain instance-specific details
+        example: A long description of the error that may contain instance-specific details
       links:
         properties:
           about:
             type: string
-            format: url
+            format: uri
             description: A link to further information about the error
             example: https://developer.oregonstate.edu/documentation/error-reference#1234
-  ErrorResult:
+  SelfLink:
     properties:
-      errors:
+      self:
+        type: string
+        format: url
+        description: Self link of resource
+  RelationshipResource:
+    description: an resource passed to a /relationships endpoint with only a type and ID
+    properties:
+      type:
+        type: string
+        description: the type of the request (pizza, dough, ingredient)
+        example: (dough|ingredient|pizza)
+      id:
+        type: string
+        description: the unique ID of the resource
+        example: abc123
+  RelationshipResourceResponse:
+    description: a response for a relationships endpoint for a -to-one relationship
+    properties:
+      data:
+        type: object
+        $ref: "#/definitions/RelationshipResource"
+  RelationshipResourceList:
+    description: a list of relationship resources
+    properties:
+      data:
         type: array
         items:
-          $ref: '#/definitions/ErrorObject'
+          $ref: "#/definitions/RelationshipResource"
+parameters:
+  pizzaId:
```

Comment:
Could you refactor the id parameters for doughs and ingredients in the same way as here?